src/cipher/encoder/encoder.go

```diff
@@ -382,20 +391,35 @@ func Size(v interface{}) int {
 	return n
 }
 
-// dataSize returns the number of bytes the actual data represented by v occupies in memory.
+// isEmpty returns true if a value is "empty".
+// Only supports Slice, Map and String.
+// All other values are never considered empty.
+func isEmpty(v reflect.Value) bool {
+	t := v.Type()
+	switch t.Kind() {
+	case reflect.String:
+		return v.Len() == 0
+	case reflect.Map:
+		return v.IsNil() || v.Len() == 0
+	case reflect.Slice:
+		return v.IsNil() || v.Len() == 0
+	default:
+		return false
```

Comment:
Can check if empty for other types in this way:

```go
return reflect.DeepEqual(v.Interface(), reflect.Zero(t).Interface())
```

Reply:
Do we have any reason to?

Reply:
Really only meant for []byte, this omitempty tag is a hack to allow us to append a field temporarily

Reply:
I know this is already merged , but I'd be in favor of having this added for all other types . Support limited to slices leads to [re-implementing encoder behavior](https://github.com/iketheadore/skycoin/blob/da40449/src/daemon/messages.go#L275-L289) .